exercises/simple-linked-list/src/test/scala/SimpleLinkedListTest.scala

```diff
@@ -0,0 +1,108 @@
+import org.scalacheck.{Arbitrary}
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+
+class SimpleLinkedListTest extends FlatSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  private implicit def arbitrarySimpleLinkedList[T](implicit arbitraryTs: Arbitrary[Array[T]]): Arbitrary[SimpleLinkedList[T]] =
+    Arbitrary {
+      arbitraryTs.arbitrary map (SimpleLinkedList(_:_*))
+    }
+
+  it should "handle single item list" in {
+    val list = SimpleLinkedList().add(1)
+    list.isEmpty should be (false)
+    list.value should be (1)
+  }
+
+  it should "handle single item has no next item" in {
+    pending
+    val list = SimpleLinkedList().add(1)
+    list.next.isEmpty should be (true)
+  }
+
+  it should "handle two item list" in {
+    pending
+    val list = SimpleLinkedList().add(1).add(2)
+    list.value should be (1)
+    list.next.value should be (2)
+  }
+
+  it should "handle two item list has no next value" in {
+    pending
+    val list = SimpleLinkedList().add(1).add(2)
+    list.next.next.isEmpty should be (true)
+  }
+
+  it should "allow creation from a Seq" in {
+    pending
+    val list = SimpleLinkedList.fromSeq(List(3, 2, 1))
```

Comment:
I think `fromSeq` should be added in the template's companion object? After all it is then part of the public API like the `trait`'s functions.

Another question is whether we should also keep the constructors in the tests or only have `fromSeq`.
The constructors make the exercise more interesting, but also somewhat harder.
A compromise could be to have the `apply` method in the template and have it calling `fromSeq`.
(see also my comment about this topic a couple of days ago).